src/modules/hardwareWallet/hooks/useHWAccounts.js

```diff
@@ -1,74 +1,57 @@
-import { useDispatch, useSelector } from 'react-redux';
+import { useSelector } from 'react-redux';
 import {
   selectCurrentHWDevice,
   selectHWAccounts,
 } from '@hardwareWallet/store/selectors/hwSelectors';
-import { setHWAccounts } from '@hardwareWallet/store/actions';
 import { getNameFromAccount } from '@hardwareWallet/utils/getNameFromAccount';
-import { getHWAccounts } from '@hardwareWallet/utils/getHWAccounts';
-import { resetLedgerIPCQueue } from '@libs/hardwareWallet/ledger/ledgerLiskAppIPCChannel/clientLedgerHWCommunication';
+import { getMultipleAddresses } from '@libs/hardwareWallet/ledger/ledgerLiskAppIPCChannel/clientLedgerHWCommunication';
 
-const useHWAccounts = () => {
-  const hwAccounts = useSelector(selectHWAccounts);
+function useHWAccounts(nrOfAccounts) {
   const currentHWDevice = useSelector(selectCurrentHWDevice);
-  const [isLoadingHWAccounts, setIsLoadingHWAccounts] = useState(false);
+  const persistedHwAccounts = useSelector(selectHWAccounts);
+  const [hwAccounts, setHwAccounts] = useState([]);
+  const [isLoading, setIsLoading] = useState(undefined);
   const [loadingHWAccountsError, setLoadingHWAccountsError] = useState();
 
-  const getAccountName = (address) => getNameFromAccount(address, hwAccounts);
-
-  const dispatch = useDispatch();
-  const { ipc } = window;
-
   useEffect(() => {
-    let isMounted = true;
-    function getUniqueAccounts(accounts) {
-      return accounts.reduce((accum, account) => {
-        const indexOfAccount = accum.findIndex(
-          (item) => item.metadata.address === account.metadata.address
-        );
-        if (indexOfAccount === -1) {
-          accum.push(account);
-        }
-        return accum;
-      }, []);
-    }
-
-    if (ipc && dispatch && currentHWDevice?.path && !isLoadingHWAccounts) {
-      // eslint-disable-next-line max-statements
+    if (nrOfAccounts > 0) {
       (async () => {
-        setIsLoadingHWAccounts(true);
+        setIsLoading(true);
         try {
-          const accounts = await getHWAccounts(currentHWDevice, getAccountName);
-          const uniqueAccounts = getUniqueAccounts(accounts);
-          if (isMounted) {
-            dispatch(setHWAccounts(uniqueAccounts));
-            setLoadingHWAccountsError(undefined);
-          }
+          const accountIndexes = Array.from(Array(nrOfAccounts).keys());
+          const addressesAndPubkeys = await getMultipleAddresses(
+            currentHWDevice.path,
+            accountIndexes
+          );
+          const accounts = Object.values(addressesAndPubkeys).map((addressAndPubkey, index) => {
+            const { address, pubKey } = addressAndPubkey;
+            const accountIndex = index + 1;
+            const name =
+              getNameFromAccount(address, persistedHwAccounts) || `Account ${accountIndex}`;
+
+            return {
+              hw: currentHWDevice,
+              metadata: {
+                address,
+                pubkey: pubKey,
+                accountIndex,
```

Comment:
```suggestion
                accountIndex: index,
```
otherwise the index used while signing will be wrong.

Reply:
Nice catch! 
Btw, do you think the default name for the first account should be Account 0 or Account 1?

Reply:
Account 1 is fine i think